group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1094,58 +1087,54 @@ private void cancelConsumerGroupSessionTimeout(
     }
 
     /**
-     * Schedules a revocation timeout for the member.
+     * Schedules a rebalance timeout for the member.
      *
-     * @param groupId               The group id.
-     * @param memberId              The member id.
-     * @param revocationTimeoutMs   The revocation timeout.
-     * @param expectedMemberEpoch   The expected member epoch.
+     * @param groupId           The group id.
+     * @param expectedMember    The expected member.
      */
-    private void scheduleConsumerGroupRevocationTimeout(
+    private void scheduleConsumerGroupRebalanceTimeout(
         String groupId,
-        String memberId,
-        long revocationTimeoutMs,
-        int expectedMemberEpoch
+        ConsumerGroupMember expectedMember
     ) {
-        String key = consumerGroupRevocationTimeoutKey(groupId, memberId);
-        timer.schedule(key, revocationTimeoutMs, TimeUnit.MILLISECONDS, true, () -> {
+        String key = consumerGroupRebalanceTimeoutKey(groupId, expectedMember.memberId());
+        timer.schedule(key, expectedMember.rebalanceTimeoutMs(), TimeUnit.MILLISECONDS, true, () -> {
             try {
                 ConsumerGroup group = getOrMaybeCreateConsumerGroup(groupId, false);
-                ConsumerGroupMember member = group.getOrMaybeCreateMember(memberId, false);
+                ConsumerGroupMember member = group.getOrMaybeCreateMember(expectedMember.memberId(), false);
 
-                if (member.state() != ConsumerGroupMember.MemberState.REVOKING ||
-                    member.memberEpoch() != expectedMemberEpoch) {
-                    log.debug("[GroupId {}] Ignoring revocation timeout for {} because the member " +
-                        "state does not match the expected state.", groupId, memberId);
+                if (!member.assignmentEquals(expectedMember)) {
```

Comment:
So this works because we removed the pending assignment fields from the equality check? In other words, this should only be false when the fields that change indicate the assignment is no longer valid and not that the assignment just failed to complete.

Reply:
I also wasn't sure about the states being the same. Is it possible when we are unstable we will be not equal to the expected member? Or am I not understanding when we transition to stable correctly.